### DISCUSSION TITLE
Push forward 1.24 Release date

### DISCUSSION THREAD

**Discussion by cici37** [kubernetes/sig-release#1877] [General]
Since we are waiting for a fix([issue detail](https://github.com/kubernetes/kubernetes/issues/108910)) in go1.18.1, and go1.18.1 release was pushed to Tuesday, April 12th([GO Announcement](https://groups.google.com/g/golang-dev/c/-6gzljWJb0k/m/REOP9ejdBwAJ)), here is the discussion on how we should push 1.24 release dates.

Due to the [slack discussion](https://kubernetes.slack.com/archives/C2C40FMNF/p1649348262088339) in sig-release channel, the currently propose is to push:
- 1.24.rc.0 cut to: Friday 15th
- 1.24.0 cut to:  27th April

Please feel free to leave your opinion.

**Comment by puerco** [comment]
OK, one con to cut it on April 14th is that I see the time window to get the go bump ready. Depending on the time the images are available we could end up having less than a day to do it. 

This means that the release could slip to Friday which we generally try to avoid. Also, it falls right on the Easter holidays, so I'm wondering about the availability of folks around those dates.

**Reply by puerco** [reply]
Obviously, the main ones are the branch managers (@palnabarun @jimangel ), but rc.0 implies also the branch cut so we need to ensure @kubernetes/release-engineering people are around too to support the BMs if something comes up.

**Reply by Verolop** [reply]
I am available to help with the releases. Including Fridays, Easter break, etc. Happy to coordinate, if necessary @palnabarun @jimangel

**Reply by saschagrunert** [reply]
@Verolop I appreciate your dedication to this project! 

Still, we should avoid cutting the release on Friday especially when we have a public holiday in many countries where nobody else is around.

**Comment by cici37** [comment]
cc @kubernetes/release-team-leads for awareness

**Comment by puerco** [comment]
Another point to consider: Regardless of when we cut rc.0 we are looking at a delay of _one week at least_.  We need to allow time for downstream users to run their tests against the rc. This is the week starting Monday 18:
```
        April 2022     
   Su Mo Tu We Th Fr Sa
                   1  2
    3  4  5  6  7  8  9
   10 11 12 13 14 15 16
 > 17 18 19 20 21 22 23
   24 25 26 27 28 29 30
```

**Comment by justaugustus** [comment]
(From https://kubernetes.slack.com/archives/C2C40FMNF/p1649693730917389?thread_ts=1649348262.088339&cid=C2C40FMNF)

@kubernetes/release-team-leads @kubernetes/release-managers @dims @liggitt -- All, after discussing with @kubernetes/sig-release-leads, let's jump back into this thread and discuss a notional delayed release date of Tuesday, May 3rd.

A few points to consider:
- We theoretically have a RC 0 scheduled for today, which won't happen due to release manager availability
- We need to apply some fixes to the branch creation jobs
- We're awaiting the go1.18.1 release (scheduled for tomorrow)
- Even though go1.18.1 is scheduled for tomorrow, they usually release mid-to-EOD US Eastern, which means:
- Optimistically, the "upstream" Golang images will not be available until Wednesday
- Add in a day+ for building our Golang images puts us at Thursday/Friday and:
- Friday/weekend takes us into Easter, for those who celebrate
- EDIT: We should not release on Monday/Friday, so Tuesday, April ~12th~ 19th, optimistically, for RC 0
- EDIT: Soak for a week and drop RC 1 on Tuesday, April ~19th~ 26th, which:
- EDIT: Leaves us ~two~ 1 week ~s~ til May 3rd ~(which includes a week of buffer in case we need to cut another RC)~

WDYT?

**Reply by justaugustus** [reply]
From @jeremyrickard:

> RC 0 == Not Go 1.18.1
> RC 1 == Go 1.18.1

I'm realizing I got the dates wrong in the original post...

RC 0 - April 19th - go1.18.1
RC 1 - April 26th
Official - May 3rd

**Reply by jeremyrickard** [reply]
OK that makes more sense :)

**Reply by liggitt** [reply]
if all the stars align, it seems like we _might_ be able to pick up a day and get RC0 out Thursday, but I don't know that that would actually move the needle much on the RC1 / .0 schedule, so the plan outlined here seems reasonable to me

**Reply by JamesLaverack** [reply]
I put a [full writeup of my thoughts in a bigger post](https://github.com/kubernetes/sig-release/discussions/1877#discussioncomment-2546211) (because I forgot how GitHub Discussions work), but TL;DR is that I am a 👍 on this schedule and a delay to Tuesday 3rd May.

**Comment by JamesLaverack** [comment]
I agree with a two-week delay to Tuesday 3rd May for the 1.24 release date.

I'd like to avoid *multiple* delays. If we're going to delay we should do it once and commit. I also don't want us to rush anything, which might result in rapidly needing to ship a 1.24.1 (or 😱 1.24.2) patch release. Both of these would erode community confidence in our process IMO.

I'm aware of the impact that a two-week delay could have on the release team as a whole. In our Monday 11th April burndown meeting, I already mentioned to the team that it's okay if people can't stay on past the original 15-week commitment and to speak to @jrsapi or myself if they have concerns.

I also agree with *not* cutting a release on a Monday or Friday.

I think cutting an rc.0 without the Go 1.18.1 fix is a bit pointless, so I'm in agreement that it should wait until Go 1.18.1 until rc.0.

There's a possible impact on 1.25 (and further) releases, cc @cici37 and @reylejano (1.25 Lead and EA). I'm unsure if it's safe to do some 1.25 spin-up tasks while 1.24 is concluding. Or if it's best to conclude 1.24 completely and push back 1.25 a bit too?

I'm also concious that this will mean that k/k's master branch will remain frozen for longer. Which may delay some folk from merging things. In general though, I don't intend to permit any exception requests for bugs that are not **critical** at this stage, even with the extension.

I'd also like to wait until the Go 1.18.1 release *actually drops* and we confrm our fix is in there, before we communicate out. If Go delays again we'll have to revise this schedule.

**Comment by dims** [comment]
@justaugustus @kubernetes/release-team @liggitt - golang folks are still on for 1.18.1 tomorrow ( https://groups.google.com/g/golang-nuts/c/9A-3jOOW4VI/m/10oNeQ93FwAJ )

@JamesLaverack the fix we were waiting on has already merged in golang 1.18 branch. so yes we should see it in 1.18.1 tomorrow.

**Comment by JamesLaverack** [comment]
Given this discussion, and after some debate in the Tuesday 12th April 2022 SIG Release Engineering meeting, as Release Lead I'm resolving to implement Stephen's plan above and ℹ️ **delay the Kubernetes 1.24 scheduled release date to Tuesday 3rd May 2022** ℹ️ .

Going forwards we should:

- [x] Update the release schedule @Verolop 
- [x] Send an email to dev@kubernetes.io announcing the change. @JamesLaverack 
- [x] Speak to the comms group about doing wider communication around the change. @JamesLaverack 
- [x] Loop in CNCF comms to arrange the webinar correctly @mickeyboxell

**Reply by JamesLaverack** [reply]
Here is my email draft, reviews appreciated: https://docs.google.com/document/d/1E8vRmYLhgVsP340jTokixBPx5xV3BnBH0kmHqVwsuCs/edit?usp=sharing

**Reply by saschagrunert** [reply]
> Here is my email draft, reviews appreciated: https://docs.google.com/document/d/1E8vRmYLhgVsP340jTokixBPx5xV3BnBH0kmHqVwsuCs/edit?usp=sharing

LGTM @JamesLaverack
